UsageEnvironment/include/Log.hh

```diff
@@ -42,13 +42,13 @@ private:
   void  PrepNextLog();
   void  LogWithLevel(LogLevel level, const std::string filePath, int line, const char* format, va_list argptr);
 public:
-  _Log(const std::string logDirectory = "");
+  _Log(const std::string logDirectory);
+  _Log();
   ~_Log() {}
 
   void BackupLog(const std::string backupFileName);
-  void OutputToFile(const std::string filePath);
-  void OutputToStdout(bool toStdout);
   void SetLevel(LogLevel level);
+  void SetTmpLogDir(std::string logDirectory);
```

Comment:
Is it intentional to allow the client to change the log directory during runtime? If not, this should be a private function so client can't call it.

Reply:
Yes it's intentional. Changing it during runtime is what allows us to enable backing up and logging to a tmp directory. I left the compile-time flag in there just in case anyone wants to use it and doesn't want to use the SetTmpLogDir function.

Reply:
OK that fine, but in that case we need to protect `SetTmpLogDir()` function body with our mutex to be safe.

(I know in theory `SetTmpLogDir()` should be called before any debug prints happen, but we want to protect against future changes where things could get moved around and prints might happen concurrently with the client calling `SetTmpLogDir()`.)